src/server/getters/database.ts

```diff
@@ -25,13 +28,78 @@ export interface Dictionary {
   [key: string]: any;
 }
 
-export function queryModifier(query: Knex.QueryBuilder, defaultSortBy: string, defaultSortOrder: string, sortBy: string|null|undefined, isSortDescending: boolean|null|undefined, limit: number|null|undefined, offset: number|null|undefined): Knex.QueryBuilder {
+export function queryModifierDB(
+  query: Knex.QueryBuilder,
+  defaultSortBy: string,
+  defaultSortOrder: string,
+  sortBy: string | null | undefined,
+  isSortDescending: boolean | null | undefined,
+  limit: number | null | undefined,
+  offset: number | null | undefined,
+): Knex.QueryBuilder {
   query = query.orderBy(sortBy || defaultSortBy, sortDirection(isSortDescending, defaultSortOrder));
   if (limit != null) query = query.limit(limit);
   if (offset != null) query = query.offset(offset);
   return query;
 }
 
+function queryModifierUserland<T>(
+  db: Knex,
+  query: Knex.QueryBuilder,
+  defaultSortBy: string,
+  defaultSortOrder: string,
+  sortBy: string | null | undefined,
+  isSortDescending: boolean | null | undefined,
+  limit: number | null | undefined,
+  offset: number | null | undefined,
+  callback: GenericCallback<Array<T>>,
+): void {
+  type RowWithSort = T & {xMySorterFieldx: BigNumber};
+
+  let sortField: string = defaultSortBy;
+  let sortDescending: boolean = defaultSortOrder.toLowerCase() === "desc";
+
+  if (sortBy != null) {
+    sortField = sortBy;
+    if (typeof(isSortDescending) !== "undefined" && isSortDescending !== null) {
```

Comment:
what if they passed in `isSortDescending: false` ?

Reply:
If so, the. The nullity and undefined checks will pass and sortDescending fill be set to false on line 65.

Reply:
oop, see it now!